lib/rules/no-constant-condition.js

```diff
@@ -180,7 +181,8 @@ module.exports = {
                         isLeftShortCircuit ||
                         isRightShortCircuit;
                 }
-
+                case "NewExpression":
+                    return true;
```

Comment:
```suggestion
                    return inBooleanPosition;
```

This would be a false positive:

```js
if (new Number(x) + 1 === 2) {}
```

`function isConstant(node, inBooleanPosition)` has two modes:

* `inBooleanPosition = true` checks if the given expression has constant truthiness. For example, `typeof x` is always truthy, so `if (typeof x);` is reported as a constant condition.
* `inBooleanPosition = false` checks if the given expression has constant value (per the logic of this rule). For example, `typeof x` is not constant, so `if (typeof x === "string");` is not reported as a constant condition.